cpp/ql/src/semmle/code/cpp/Parameter.qll

```diff
@@ -153,3 +153,10 @@ class Parameter extends LocalScopeVariable, @parameter {
     )
   }
 }
+
+/**
+ * An `int` that is a parameter index for some function.  This is needed for binding in certain cases.
+ */
+class ParameterIndex extends int {
```

Comment:
Can this be replaced with `ArgumentOperandTag`, or something equivalent that can be consumed by `ArgumentOperandTag`?

Reply:
Probably. Ideally we'd come up with something that could be used across all the model interfaces as well as for `ArgumentOperandTag`.

Reply:
How about just take `ArgumentOperandTag`, including its subtypes and IPA types, rename it to `ArgumentId` or something like that, and put it in the ASTs somewhere. Then, `TArgumentOperandTag` is just `TArgumentOperandTag(ArgumentId id)` plus the necessary classes to wrap it.